doc/dev/vstart-runner.rst

```diff
@@ -0,0 +1,79 @@
+    $ cd teuthology/
+    $ virtualenv ./venv
+    $ source venv/bin/activate
+    $ pip install --upgrade pip
+    $ pip install -r requirements.txt
+    $ python setup.py develop
+    $ deactivate
+
+Above steps installs teuthology in a virtual environment.
+
+**NOTE***: The pip command above is pip2, not pip3.
+
+**NOTE***: vstart_runner.py as well as most tests in qa/ are only compatible
+        with ``python2``. Therefore, use ``python2`` to run the tests locally.
+
+In a Ceph repository that has been successfully built perform following steps
+to run one of the tests locally -::
+
+    $ cd build
+    $ source ~/path/to/teuthology/venv/bin/activate
+    $ ../src/vstart.sh -n -d -l
+    $ python2 ../qa/tasks/vstart_runner.py tasks.cephfs.test_client_recovery.TestClientRecovery.test_reconnect_timeout
+
+The last command gets the vstart_runner.py running which in turn executes the
+test `test_reconnect_timeout`_ from `TestClientRecovery`_ in
+``qa/tasks/cephfs/test_client_recovery``. However, the command assumes that the
+Ceph cluster is already up locally. To create a cluster beforehand you can
+pass ``--create`` option to vstart_runner.py.
+
+vstart_runner.py can take 3 options -
+
+--create                    create Ceph cluster before running a test
+--create-cluster-only       creates the cluster and quits; tests can be issued
+                            later
+--interactive               drops a Python shell when a test fails
+
+Internal working of vstart_runner.py -
+-----------------------------------------
+vstart_runner.py primarily does three things -
+
+* collecting and running the tests
+    vstart_runner.py setups/teardowns the cluster and collects and runs the
+    test. This is implemented using methods ``scan_tests()``,
+    ``load_tests()``, ``exec_test()``. This is where all the options that
+    vstart_runner.py takes are implemented along with other features like
+    logging and copying features to the bottom of the log.
+
+* provides an interface for issuing and testing shell commands
+    The tests are written assuming the clusters exists on a remote machines.
+    ``vstart_runner.py`` provides an interface to run the test with cluster that
+    exists within the local machine. This done in class ``LocalRemote``. The
+    process the runs ``LocalRemote`` is implemented as ``LocalRemoteProcess``.
+    Class ``LocalDaemon`` provides an interface to handle Ceph Daemons and
+    class ``LocalFuseMount`` to create and handle ``FuseMount``
+
+* provides an interface to issue Ceph cluster commands
+    This is implemented as class ``LocalCephManager``. It provides methods to
+    run a Ceph cluster command with and without admin socket.
+
+
+**NOTE***: vstart_runner.py can create mount CephFS only with FUSE. Therefore,
+        make sure that fuse package is installed and enabled on your system.
+
+**NOTE***: Make sure that ``use_allow_other`` is added to ``/etc/fuse.conf``.
```

Comment:
```suggestion
**NOTE**: Make sure that ``use_allow_other`` is added to ``/etc/fuse.conf``.
```
In other places too there is extra asterisk.

Reply:
It's so that the wrapped text appears bold.